deeppavlov/dataset_iterators/file_paths_iterator.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2017 Neural Networks and Deep Learning lab, MIPT
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import List, Tuple, Iterator, Optional
+from  collections import deque
+
+import numpy as np
+
+from deeppavlov.core.common.registry import register
+from deeppavlov.core.data.data_learning_iterator import DataLearningIterator
+from deeppavlov.core.common.log import get_logger
+
+
+log = get_logger(__name__)
+
+@register('file_paths_iterator')
+class FilePathsIterator(DataLearningIterator):
+    """Dataset iterator for datasetes like 1 Billion Word Benchmark
+
+    Args:
+        data: dict with keys ``'train'``, ``'valid'`` and ``'test'`` and values
+        seed: random seed for data shuffling
+        shuffle: whether to shuffle data during batching
+
+    Attributes:
+        shuffle: whether to shuffle data during batching
+        random: instance of ``Random`` initialized with a seed
+    # """
+
+    def __init__(self, 
+                 data: dict, 
+                 seed: Optional[int] = None, 
+                 shuffle: bool = True,
+                 *args, **kwargs) -> None:
+        self.seed = seed
+        self.np_random = np.random.RandomState(seed)
+        super().__init__(data, seed, shuffle, *args, **kwargs)
+
+    @staticmethod
+    def _chunk_generator(items_list, chunk_size):
```

Comment:
can be replaced with https://docs.python.org/3/library/itertools.html#itertools.islice

Reply:
it will be more difficult for understanding for an example look at this 
`from itertools import islice

def grouper(iterable, n):
    it = iter(iterable)
    group = tuple(islice(it, n))
    while group:
        yield group
        group = tuple(islice(it, n))  `